README.md

```diff
@@ -51,85 +51,125 @@ some kind of discriminant:
+
+ADT enums support an object literal-like construction syntax:
+
+```js
+const msg = Message.Move{ x: 10, y: 20 };
 ```
 
 # Semantics
 
 ## Well-Known Symbols
 
-This proposal introduces three new well-known symbols that are used with enums:
+This proposal introduces four new well-known symbols that are used with enums:
 
-| Specification Name | \[\[Description]] | Value and Purpose |
-|:-|:-|:-|
-| @@toEnum | `"Symbol.toEnum"` | A method that is used to derive the value for an enum member during _EnumMember_ evaluation. |
-| @@formatEnum | `"Symbol.formatEnum"` | A method of an _enum object_ that is used to convert a value into a string representation based on the member names of the enum. Called by `Enum.format`. |
-| @@parseEnum | `"Symbol.parseEnum"` | A method of an _enum object_ that is used to convert a member name String into the value represented by that member of the enum. Called by `Enum.parse`. |
+| Specification Name  | \[\[Description]]            | Value and Purpose |
+|:--------------------|:-----------------------------|:------------------|
+| @@toEnum            | `"Symbol.toEnum"`            | A method that is used to derive the value for an enum member during _EnumMember_ evaluation. |
+| @@formatEnum        | `"Symbol.formatEnum"`        | A method of an _enum object_ that is used to convert a value into a string representation based on the member names of the enum. Called by `Enum.format`. |
+| @@parseEnum         | `"Symbol.parseEnum"`         | A method of an _enum object_ that is used to convert a member name String into the value represented by that member of the enum. Called by `Enum.parse`. |
+| @@propertyConstruct | `"Symbol.propertyConstruct"` | A method of an ADT _enum member_ that is used to construct an ADT record-like enum value using object literal-like syntax. |
 
 ## Enum Declarations
 
-Enum declarations consist of a finite set of _enum members_ that define the names and values
-for each member of the enum. These results are stored as properties of an _enum object_. An 
-_enum object_ is an ordinary object with an \[\[EnumMembers]] internal slot, and whose 
-\[\[Prototype]] is `null`.
+Enum declarations consist of a finite set of _enum members_ that define the names and values for each member of the
+enum. These results are stored as properties of an _enum object_. An _enum object_ is an ordinary object with an
+\[\[EnumMembers]] internal slot, and whose \[\[Prototype]] is `null`.
 
-### Automatic Initialization
+### Initializers
 
-If an _enum member_ does not supply an _Initializer_, the value of that _enum member_ will be 
-automatically initialized:
+By default, _enum members_ must have explicit initializers:
 
 ```js
 enum DaysOfTheWeek {
+  Sunday = 0,
+  Monday = 1,
+  Tuesday, // error
+}
+```
+
+### Automatic Initialization
+
+Automatic initialization is supported by way of an _Enum Mapper Object_ provided via an `of` clause:
+
+```js
+// Number-based enum with auto-increment
```

Comment:
Is mixing of auto-incremented values and explicit allowed as in some languages e.g., C#? In that case, should this proposal be explicit in how that will work e.g., auto-increments from the previous auto or explicit value?